python/fusion_engine_client/analysis/analyzer.py

```diff
@@ -142,6 +142,9 @@ def __init__(self,
             'return_numpy': True
         }
 
+        # For plots that only need to trace one source ID.
+        self.default_source_id = min(self.reader.source_ids)
```

Comment:
Use `get_available_source_ids()` instead of accessing this member directly

Reply:
Also, if the user does specify the `source_ids` argument, you should set this to the intersection of the available IDs and the ones the user requested. If the intersection is empty, there will be no available data at all and we should print an error and abort.